xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/main/java/org/xwiki/notifications/filters/NotificationFilterManager.java

```diff
@@ -59,6 +60,23 @@
     Collection<NotificationFilter> getAllFilters(DocumentReference user, boolean onlyEnabled)
         throws NotificationException;
 
+    /**
+     * Get all notifications filters that are enabled to the given user and matching the requested filtering moment.
+     *
+     * @param user reference to the user
+     * @param onlyEnabled either or not only filters enabled for the user should be collected
+     * @param filteringMoments when the filter should be used
+     * @return the collection of notification filters enabled to the user.
+     * @throws NotificationException if an error happens
+     * @since 10.4RC1
```

Comment:
Not sure about that one :)

Reply:
Oups, copy/pasted too fast, I forgot unstable too.